writer/writer.go

```diff
@@ -205,7 +205,9 @@ func (self *ParquetWriter) flushObjs() error {
 					}
 					close(doneChan)
 				}
-				doneChan <- 0
+				if _, ok := <- doneChan; ok {
```

Comment:
it will get one value from doneChan and then put one in it.
It will cause the main routine wait forever.

Reply:
@xitongsys when it recovers from a panic, it hits line 206 where it closes the channel. Then it tries to write on the closed channel and panics. It's a pretty obvious, please do see for yourself. It happened to me today, that's why I opened this PR. All this PR does is it prevents writing in the channel if it's already closed. It may not be a perfect solution as it need to be able to read something for the channel first and that could be blocking forever. But the current logic is prone to a panic if not fixed, as explained above. However, I hope to have time to dig a little bit deeper and find why did it panic (and then recovered) in the first place.

Reply:
hi, I use sync.WaitGroup instead of doneChan to fix this issue. Please check the latest commits. Thanks.

Reply:
Hi, @xitongsys I've just had a look and it looks good to me. It tested it as well and it doesn't panic anymore. Thank you.

Reply:
One minor suggestion based on Go best practices would be to always defer call to waitGroup's Done() func.
https://github.com/xitongsys/parquet-go/blob/ae2fa1c6d032c18a6ffc271927595fba4e20a073/layout/rowgroup.go#L72
https://github.com/xitongsys/parquet-go/blob/ae2fa1c6d032c18a6ffc271927595fba4e20a073/writer/writer.go#L198 

Also, I think defering a closure is not necessary here:
https://github.com/xitongsys/parquet-go/blob/ae2fa1c6d032c18a6ffc271927595fba4e20a073/reader/reader.go#L326

For reference, please have a look at: https://golang.org/pkg/sync/#example_WaitGroup